content/blog/introduction-to-typescript/index.md

```diff
@@ -0,0 +1,212 @@
+
+Let's go over some of the arguments in favor of TypeScript's developer quality of life improvements.
+
+### Improved Tooling Support {#tooling}
+
+Historically, having the ability to make assumptions about code in order to provide developer niceties (such as autocomplete code suggestions) in loosely typed languages such as JavaScript has been incredibly hard to do. As time has gone on, support for these types of actions have gotten better; but due to the nature of JavaScript's type system, there will likely always be limitations on how effectively this can be done. TypeScript's syntax, however, _can provide much of the type data about your source code needed for tools to be able to provide those niceties_ that are otherwise tricky for these tools to build. _The TypeScript team even provides a tool to communicate directly to these IDEs_ so that the work on implementing this syntax data consumption is much more trivial than they otherwise would be. _This is why [many changelogs for TypeScript releases](https://www.typescriptlang.org/docs/handbook/release-notes/overview.html) mention changes to editors such as [Visual Studio Code](https://code.visualstudio.com)_.
+
+#### 3rd Party Library Support {#typing-files}
+
+Because of JavaScript's awesome engineering diversity, many widely used projects do not use TypeScript. However, _there are ways we can still utilize TypeScript's tooling capabilities without porting the code_. If you have a good understanding of the given project's codebase and TypeScript, _you can write a definition file which sits separated from the rest of the codebase_. These definition files allow you many of the same tooling abilities native TypeScript source code allows.
+
+An example of these typings might look something like this:
+
+```javascript
+// index.js
+function ANumberToAString(numProp) {
+	if (typeof numProp !== "number") throw "Only numbers are supported";
+	return numProp.toString();
+}
+```
+
+```typescript
+// index.d.ts
+declare function ANumberToAString(numProp: number): string; // Accept a number arg, return a string
+```
+
+##### Community Hosting {#definitely-typed}
+
+Additionally, because TypeScript has a well established and widely used install-base, **there are already many different definition files in the wild for supporting non-TypeScript supporting projects**. One of the more extensive collections of these typings lives at the [DefinitelyTyped repository](https://github.com/DefinitelyTyped/DefinitelyTyped), which publishes the typings of their typings under the package names `@types/your-package-name` (where `your-package-name` is the name of the project you're looking for typings of) that you can look for on your package manager.
```

Comment:
```suggestion
Additionally, because TypeScript has a well established and widely used install-base, **there are already many different definition files in the wild for supporting non-TypeScript supporting projects**. One of the more extensive collections of these typings lives at the [DefinitelyTyped repository](https://github.com/DefinitelyTyped/DefinitelyTyped), which publishes the package's community typings under the package names `@types/your-package-name` (where `your-package-name` is the name of the project you're looking for typings of) that you can look for on your package manager.
```